src/rosdiscover/interpreter/topic.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+__all__ = ('Topic',)
+from typing import Any, Dict
+import attr
+
+
+@attr.s(frozen=True, slots=True)
```

Comment:
This is great. If you want to make it even tidier, you can add `auto_attribs=True` to the `attr.s` call and drop ` = attr.ib()` from each attribute. (See: https://github.com/ChrisTimperley/rosdiscover/blob/master/src/rosdiscover/interpreter/summary.py#L9-L12)